src/components/Usermenu.tsx

```diff
@@ -0,0 +1,201 @@
+import { usePlaceholder } from "@/classe/User";
+import useBreakpoints from "@/utils/useBreakpoints";
+import { AccountCircle, Logout, Settings } from "@mui/icons-material";
+import {
+  Box,
+  ListItemIcon,
+  Menu,
+  MenuItem,
+  Typography,
+  useTheme,
+} from "@mui/material";
+import { ReactElement } from "react";
+// import useBreakpoints from "@/utils/useBreakpoints";
+
+/**
+ * UsermenuProps
+ *
+ * Interface for the Usermenu component
+ *
+ * @param {null | HTMLElement} anchorElement - The anchor element for the user popup
+ * @param {boolean} userPopUpOpen - The state of the user popup
+ * @param {(open: boolean) => void} setUserPopUpOpen - Function to set the user popup state
+ * @param {(anchorEl: null | HTMLElement) => void} setAnchorElement - Function to set the anchor element
+ */
+type UsermenuProps = {
+  anchorElement: null | HTMLElement;
+  userPopUpOpen: boolean;
+  setUserPopUpOpen: (open: boolean) => void;
+  setAnchorElement: (anchorEl: null | HTMLElement) => void;
+};
+
+/**
+ * UserPopup Component
+ *
+ * This component renders a User Popup which appears when the user clicks on the
+ * account icon in the header. It provides options to view the user's profile,
+ * navigate to the dashboard, and log out. It also displays the current app version.
+ *
+ * @param {UsermenuProps} props - The properties of the Usermenu component
+ * @returns {ReactElement} A Usermenu component
+ *
+ */
+const Usermenu = ({
+  anchorElement,
+  userPopUpOpen,
+  setUserPopUpOpen,
+  setAnchorElement,
+}: UsermenuProps): ReactElement => {
+  const theme = useTheme();
+  const { isDownXs, isBetweenXsSm} = useBreakpoints();
+  const placeholderUser = usePlaceholder();
+
+  /**
+   * Function to handle closing the user popup
+   */
+  const handleClose = (): void => {
+    setUserPopUpOpen(false);
+    setAnchorElement(null);
+  };
+
+  /**
+   * Prevents the default behavior of the event
+   */
+  const preventDefault = (event: React.MouseEvent<HTMLLIElement, MouseEvent>): void => {
+    event.preventDefault();
+  };
```

Comment:
See below, you can use this inline instead of as a separate function since it is not used in multiple places